tests/schema_test.go

```diff
@@ -15,49 +15,58 @@
 package tests
 
 import (
+	"bytes"
 	"encoding/json"
+	"fmt"
 	"io/ioutil"
+	"os"
 	"testing"
 
-	"github.com/pkg/errors"
-	"github.com/pulumi/crd2pulumi/gen"
+	"github.com/pulumi/crd2pulumi/pkg/codegen"
 	pschema "github.com/pulumi/pulumi/pkg/v3/codegen/schema"
 	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/yaml"
 )
 
 const TestCombineSchemasYAML = "test-combineschemas.yaml"
 const TestGetTypeSpecYAML = "test-gettypespec.yaml"
 const TestGetTypeSpecJSON = "test-gettypespec.json"
 
-func UnmarshalSchemas(yamlPath string) (map[string]interface{}, error) {
-	yamlFile, err := ioutil.ReadFile(yamlPath)
-	if err != nil {
-		return nil, errors.Wrapf(err, "could not read file %s", yamlPath)
+// UnmarshalYaml un-marshals one and only one YAML document from a file
+func UnmarshalYaml(yamlFile []byte) (map[string]interface{}, error) {
+	dec := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(yamlFile), 128)
+	var value map[string]interface{}
+	if err := dec.Decode(&value); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal YAML: %w", err)
 	}
-	schema, err := gen.UnmarshalYaml(yamlFile)
+	return value, nil
+}
+
+func UnmarshalSchemas(yamlPath string) (map[string]interface{}, error) {
+	yamlFile, err := os.ReadFile(yamlPath)
 	if err != nil {
-		return nil, errors.Wrapf(err, "could not unmarshal %s", yamlPath)
+		return nil, err
 	}
-	return schema, nil
+	return UnmarshalYaml(yamlFile)
 }
```

Comment:
Are these functions needed here, or can this logic be imported? This looks similar to the logic used in the CLI.